packages/cfd/src/features/hooks/useCurrentList.ts

```diff
@@ -0,0 +1,51 @@
+import { useStore } from '@deriv/stores';
+import { useMT5AccountsList, useDxtradeAccountsList, useCtraderAccountsList } from '@deriv/api';
+import { getAccountListKey } from '@deriv/shared';
+
+type MT5AccountsList = NonNullable<ReturnType<typeof useMT5AccountsList>['data']>[number];
+type DxtradeAccountsList = NonNullable<ReturnType<typeof useDxtradeAccountsList>['data']>[number];
+type CtraderAccountsList = NonNullable<ReturnType<typeof useCtraderAccountsList>['data']>[number];
+
+type TCurrentList = {
+    [key: string]: MT5AccountsList | DxtradeAccountsList | CtraderAccountsList;
+};
+
+/* 
+    This hook is used to get the existing_accounts_data
+*/
+const useCurrentList = () => {
```

Comment:
But i have a question in general. Why we doing this? We can just call the hooks in the components we wanna use and be done with it. Why another transformation ? 🤔

Reply:
Currently we are facing issue with the Context that which account we need from the response. Moreover in legacy code we ae using it like mt5.synthetic_all@ps+01 as the key for the account response. Thats why we are transforming to the legacy code because we cannot use Context as we want to separate CFD from Appstore. The context can be provided in Appstore but how we use it inside the CFD. If we pass it again we are introducing a new dependency.

If you have any solution please bring it on we will be happy to have that as our main goal is to have a better code in the deriv-app

Reply:
In addition to what Hamza mentioned, we need to stick with the current logic that we have, add related hooks to do the removal of stores first and then decide if refactoring those logics are necessary. This is the main reason of adding some hooks to the CFD package.